src/structures/Sticker.js

```diff
@@ -0,0 +1,107 @@
+'use strict';
+
+const Base = require('./Base');
+const { StickerFormatTypes } = require('../util/Constants');
+const Snowflake = require('../util/Snowflake');
+
+/**
+ * Represents a Sticker.
+ * @extends {Base}
+ */
+class Sticker extends Base {
+  constructor(client, sticker) {
+    super(client);
+    /**
+     * The ID of the sticker
+     * @type {Snowflake}
+     */
+    this.id = sticker.id;
+
+    /**
+     * The ID of the sticker's image
+     * @type {string}
+     */
+    this.asset = sticker.asset;
+
+    /**
+     * The description of the sticker
+     * @type {string}
+     */
+    this.description = sticker.description;
+
+    /**
+     * The format of the sticker
+     * @type {string}
+     */
+    this.format = StickerFormatTypes[sticker.format_type];
+
+    /**
+     * The name of the sticker
+     * @type {string}
+     */
+    this.name = sticker.name;
+
+    /**
+     * The ID of the pack the sticker is from
+     * @type {Snowflake}
+     */
+    this.packID = sticker.pack_id;
+
+    /**
+     * The ID of sticker preview image
+     * @type {?string}
+     */
+    this.previewAsset = sticker.preview_asset;
+
+    /**
+     * A comma-separated list of tags for the sticker
+     * @type {?string}
+     */
+    this.tags = sticker.tags;
+  }
+
+  /**
+   * A link to the sticker.
+   * <warn>If format is LOTTIE, it returns the URL of the Lottie json file.
+   * Lottie json files must be converted to display in discord.</warn>
+   * @param {StickerURLOptions} [options={}] Options for the Image URL
+   * @returns {?string}
+   */
+  stickerURL({ size } = {}) {
+    return this.client.rest.cdn.Sticker(this.id, this.asset, this.format, size);
+  }
+
+  /**
+   * A link to the sticker preview image.
+   * <warn>If format is LOTTIE, it returns the URL of the Lottie json file.
+   * Lottie json files must be converted to display in discord.</warn>
+   * @param {StickerURLOptions} [options={}] Options for the Image URL
+   * @returns {?string}
+   */
+  stickerPreviewURL({ size } = {}) {
+    if (!this.previewAsset) return null;
+    return this.client.rest.cdn.Sticker(this.id, this.previewAsset, this.format, size);
```

Comment:
Take the URL `https://media.discordapp.net/stickers/748293342357356564/17e05030dcafa6604ee7479789652459.png?size=256&passthrough=false`

![](https://media.discordapp.net/stickers/748293342357356564/17e05030dcafa6604ee7479789652459.png?size=256&passthrough=false)

Setting `passthrough` to false seems to make the sticker static, while setting it to true or omitting it seems to make it animate

![](https://media.discordapp.net/stickers/748293342357356564/17e05030dcafa6604ee7479789652459.png?size=256&passthrough=true)

Do we want to document this? Seems like there's the `discord.com/stickers/<skuID>/stickerID` URL and then there's the preview one from media? Seems like they're different URLs altogether (between asset and previewAsset), so this might need further research and another CDN route

Reply:
For now, I added `passthrough` option at new function `makeStickerUrl`.

When we discover previewAsset's CDN I will add it.